partiql-lang/src/test/kotlin/org/partiql/lang/planner/transforms/PartiQLSchemaInferencerTests.kt

```diff
@@ -991,63 +1010,296 @@ class PartiQLSchemaInferencerTests {
+                        ([1, 2] <= (SELECT id, id + 1 FROM aws.ddb.pets)) AS lit_array_lte_ids  -- id is INT
+                    FROM
+                        << 0, 1, 2 >> AS t
+                """.trimIndent(),
+                expected = BagType(
+                    StructType(
+                        fields = listOf(
+                            StructType.Field("some_str", STRING),
+                            StructType.Field("concat_str", STRING),
+                            StructType.Field("one_lt_id", BOOL),
+                            StructType.Field("id_gt_one", BOOL),
+                            StructType.Field("one_in_ids", BOOL),
+                            StructType.Field("array_in_ids", BOOL),
+                            StructType.Field("lit_array_lte_ids", BOOL),
+                        ),
+                        contentClosed = true,
+                        constraints = setOf(
+                            TupleConstraint.Open(false), TupleConstraint.UniqueAttrs(true), TupleConstraint.Ordered
+                        )
+                    )
+                )
+            ),
+            ErrorTestCase(
+                name = "List of Lists on RHS for IN",
+                query = "1 IN (SELECT id, id + 1 FROM aws.ddb.pets)",
+                expected = BOOL,
+                problemHandler = assertProblemExists {
+                    Problem(
+                        sourceLocation = UNKNOWN_PROBLEM_LOCATION,
+                        details = SemanticProblemDetails.IncompatibleDatatypesForOp(
+                            listOf(INT, BagType(ListType(INT))),
+                            "IN"
+                        )
+                    )
+                }
+            ),
+            ErrorTestCase(
+                name = "Coercion of select with multiple projections in COALESCE",
+                query = "COALESCE(SELECT id AS id, id + 1 AS increment FROM aws.ddb.pets)",
+                problemHandler = assertProblemExists {
+                    Problem(
+                        sourceLocation = UNKNOWN_PROBLEM_LOCATION,
+                        details = SemanticProblemDetails.CoercionError(
+                            StructType(
+                                fields = listOf(
+                                    StructType.Field("id", INT),
+                                    StructType.Field("increment", INT),
+                                ),
+                                contentClosed = true,
+                                constraints = setOf(
+                                    TupleConstraint.Open(false), TupleConstraint.UniqueAttrs(true), TupleConstraint.Ordered
+                                )
+                            ),
+                        )
+                    )
+                }
+            ),
+            ErrorTestCase(
+                name = "Lists on LHS for IN",
+                query = "[1, 2] IN (SELECT id FROM aws.ddb.pets)",
```

Comment:
curious would a list of ints on the LHS (List<Int>) and list of list of ints on RHS (List<List<Int>>) be valid? I would assume so

Reply:
Yes, this test case is shown in test with name `"Subquery coercion in SFW"`